src/main/java/io/strimzi/kafka/bridge/http/HttpBridge.java

```diff
@@ -303,24 +326,65 @@ public void stop(Promise<Void> stopPromise) {
         // admin client cleanup
         this.httpBridgeContext.closeHttpAdminClientEndpoint();
 
-        if (this.httpServer != null) {
+        Future<Void> apiServerShutdown = this.apiServer != null ?
+                this.apiServer.shutdown()
+                        .onFailure(t -> LOGGER.info("Error while shutting down HTTP bridge server"))
+                : Future.succeededFuture();
 
-            this.httpServer.shutdown()
-                    .onSuccess(v -> {
-                        LOGGER.info("HTTP-Kafka bridge has been shut down successfully");
-                        stopPromise.complete();
-                    })
-                    .onFailure(t -> {
-                        LOGGER.info("Error while shutting down HTTP-Kafka bridge", t);
-                        stopPromise.fail(t);
-                    });
-        }
+        Future<Void> managementServerShutdown = this.managementServer != null ?
+            this.managementServer.shutdown()
+                    .onFailure(t -> LOGGER.info("Error while shutting down HTTP bridge management server"))
+            : Future.succeededFuture();
+
+        Future.all(apiServerShutdown, managementServerShutdown)
```

Comment:
Should we use `join` here instead so we at least wait for both to finish trying to shutdown before returning?